mqtt/mqtt-broker/src/broker.rs

```diff
@@ -100,6 +104,44 @@ where
         Ok(self.snapshot())
     }
 
+    fn reevaluate_subscriptions(&mut self) -> Result<(), Error> {
+        let mut client_ids_to_remove: Vec<ClientId> = Vec::new();
+        for session in &self.sessions {
+            for sub in session.1.subscriptions().clone().unwrap().values() {
```

Comment:
as a rule, we must not use `.unwrap()` in prod code. You need to filter our Disconnecting sessions. You can probably use `filter` combinator for it, or match. Whatever feels better for you.